File: inject_loading.py
import os
import glob

html_files = glob.glob('*.html')

head_injection = '    <link rel="stylesheet" href="css/loading.css">\n'
scripts_injection = '''    <!-- Contextual Loading System -->
    <script src="js/loading/loading-messages.js"></script>
    <script src="js/loading/loading-manager.js"></script>
    <script src="js/loading-screen.js"></script>
'''

for file in html_files:
    with open(file, 'r', encoding='utf-8') as f:
        content = f.read()

    modified = False

    # 1. Inject CSS
    if 'css/loading.css' not in content:
        # Find closing </head>
        content = content.replace('</head>', head_injection + '</head>')
        modified = True

    # 2. Inject Scripts before page-transition.js
    if 'js/loading/loading-manager.js' not in content:
        # Replace <script src="js/page-transition.js">
        content = content.replace(
            '<script src="js/page-transition.js"></script>',
            scripts_injection + '    <script src="js/page-transition.js"></script>'
        )
        modified = True

    if modified:
        with open(file, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Updated {file}")

File: inject_loading.js
const fs = require('fs');
const glob = require('fs').readdirSync('.');

const htmlFiles = glob.filter(file => file.endsWith('.html'));

const headInjection = '    <link rel="stylesheet" href="css/loading.css">\n</head>';
const scriptsInjection = `    <!-- Contextual Loading System -->
    <script src="js/loading/loading-messages.js"></script>
    <script src="js/loading/loading-manager.js"></script>
    <script src="js/loading-screen.js"></script>
    <script src="js/page-transition.js"></script>`;

htmlFiles.forEach(file => {
    if (file === 'index.html') return; // already did CSS for index.html, need to do script though
    let content = fs.readFileSync(file, 'utf-8');
    let modified = false;

    // 1. Inject CSS
    if (!content.includes('css/loading.css')) {
        content = content.replace('</head>', headInjection);
        modified = true;
    }

    // 2. Inject Scripts
    if (!content.includes('js/loading/loading-manager.js')) {
        if (content.includes('<script src="js/page-transition.js"></script>')) {
            content = content.replace(
                '<script src="js/page-transition.js"></script>',
                scriptsInjection
            );
            modified = true;
        }
    }

    if (modified) {
        fs.writeFileSync(file, content, 'utf-8');
        console.log('Updated ' + file);
    }
});

// For index.html, script specifically:
let indexContent = fs.readFileSync('index.html', 'utf-8');
if (!indexContent.includes('js/loading/loading-manager.js')) {
    indexContent = indexContent.replace(
        '<script src="js/loading-screen.js"></script>',
        ''
    ).replace(
        '<script src="js/page-transition.js"></script>',
        scriptsInjection
    );
    fs.writeFileSync('index.html', indexContent, 'utf-8');
    console.log('Updated index.html scripts');
}
